includes/class-wc-skroutz-analytics-product.php

```diff
@@ -111,4 +118,31 @@ private function get_parent_product() {
 
 		return wc_get_product( $parent_id );
 	}
+
+	/**
+	 * Get the suffix of the product id based on the grouped attributes settings.
+	 * Format: -product_term_id_1-product_term_id_2...
+	 *
+	 * @return NULL|string The product id suffix. If no product attribute matches any attribute in settings returns null.
+	 *
+	 * @since    1.5.0
+	 * @access   private
+	 */
+	private function get_product_terms_suffix() {
+		$product_attributes = $this->product->get_attributes();
+
+		$terms = null;
+		foreach ($this->items_product_id_settings['grouping_attributes'] as $attribute_name) {
+			$taxonomy_name = wc_attribute_taxonomy_name($attribute_name);
+
+			if ( isset($product_attributes[$taxonomy_name]) ) {
+				$term = get_term_by( 'slug', $product_attributes[$taxonomy_name], $taxonomy_name );
+				if ( $term ) {
+					$terms .= "-{$term->term_id}";
+				}
```

Comment:
would it be better if we used the following?
`if ( $term ) $terms .= "-{$term->term_id}";`